ebean-datasource/src/main/java/io/ebean/datasource/pool/PooledConnection.java

```diff
@@ -350,12 +385,21 @@ public PreparedStatement prepareStatement(String sql, int resultSetType, int res
    * Reset the connection for returning to the client. Resets the status,
    * startUseTime and hadErrors.
    */
-  void resetForUse() {
+  void resetForUse() throws SQLException {
     this.status = STATUS_ACTIVE;
     this.startUseTime = System.currentTimeMillis();
     this.createdByMethod = null;
     this.lastStatement = null;
     this.hadErrors = false;
+    // CHECKME: Shoud we keep the asserts here or should we even reset schema/catalog here
```

Comment:
Can we rely that a connection is successfully reset when it was put back to the pool, or should we play it safe here and enforce the correct schema/catalog here

Reply:
> Can we rely that a connection is successfully reset when ...

Yes. That is in a try catch and that can fail, and if it fails its force closed and so doesn't actually make it back into the pool. 

So yes we need to do the schema reset etc when returning the connection because that can fail and a "force close" on the connection (so the connection effectively does not make it back into the pool).

> enforce the correct schema/catalog here

I'm saying that is not preferred as we have already obtained this connection from the pool at this point and are just about to give it to the application. _Needing_ to do things here "on borrow" is problematic and we really want to do those things "on return".

In this sense I think we remove these asserts (we don't want any side effects here).